doc/users-guide/base/configs.rst

```diff
@@ -0,0 +1,220 @@
+
+.. literalinclude:: examples/users-guide/base/configs.execopts
+  :language: text
+  :lines: 2
+
+There is no difference between these two forms—it's simply a matter of
+taste.
+
+As with normal variables, a variable declared as a ``config var`` can
+be re-assigned throughout its lifetime.  In contrast, ``config const``
+declarations cannot be changed once they are initialized, whether that
+initial value comes from the source code expression or the
+command-line.  The following statement declares a pair of configurable
+constants:
+
+.. literalinclude:: examples/users-guide/base/configs.chpl
+  :language: chapel
+  :lines: 2-3
+
+As with the previous case, we can override their default values on the
+executable's command-line.  For instance the following switches to a
+smaller value of *epsilon* and flips *debug* to ``true``:
+
+.. literalinclude:: examples/users-guide/base/configs.execopts-fake
+  :language: text
+  :lines: 3
+
+Boolean ``config`` declarations, like *debug* above, also support a
+mode in which they can be set to ``true`` simply by using the flag
+without a supporting value.  Thus, the previous command-line could
+also have been written:
+
+.. literalinclude:: examples/users-guide/base/configs.execopts-fake
+  :language: text
+  :lines: 4
+
+
+``config param`` and ``config type``
+------------------------------------
+
+Chapel also supports configurable ``param`` and ``type`` declarations
+via command-line flags.  However, since ``param`` values and types
+must be known to the compiler, the default values are overridden using
+flags on the compiler rather than the generated executable.
+
+As an example, the following statements declare a series of
+configurable ``param`` and ``type`` symbols:
+
+.. literalinclude:: examples/users-guide/base/configs2.chpl
+  :language: chapel
+  :lines: 3-7
+
+When overriding ``config`` declarations in the compiler the ``-s``
+flag is used to set the new value.  As an example, to set **verbose**
+to ``true``, we could compile as follows:
+
+.. literalinclude:: examples/users-guide/base/configs2.compopts-fake
+  :language: text
+  :lines: 1
+
+As with the generated executable, boolean values can also be set on
+the compiler's command line simply using the name of the flag.  Thus,
+the following would be equivalent to the previous line:
+
+.. literalinclude:: examples/users-guide/base/configs2.compopts-fake
+  :language: text
+  :lines: 2
+
+Needless to say, multiple ``config`` declarations can be overridden
+at once.  For example, this command-line overrides the default
+initializers for *bitMask*, *size*, and *age*:
+
+.. literalinclude:: examples/users-guide/base/configs2.compopts-fake
+  :language: text
+  :lines: 3
+
+One of the nice things about setting ``config`` declarations in the
+compiler is that the specified value can take the form of reasonably
+arbitrary Chapel code since the compiler is involved in interpreting
+it.  Thus, we can use more interesting expressions when setting
```

Comment:
The comments on this paragraph are good.  I went with the following, if you guys want to give it another look:

    One of the nice things about setting ``config`` declarations on the
    compiler's command-line is that the specified value can be an
    arbitrary Chapel expression, provided that the compiler can evaluate
    it in place of the source-based initialization	expression.

Since we go on to say that config consts and vars can be set on the compiler line as well, I tried to stay away from anything as specific as saying that the compiler had to be able to evaluate the expressions and instead focused on the expression needing to make sense in the context of the initializer (which arguably subsumes the compile-time condition for params and types while also saying something stronger about what is necessary for the expression to be legal).

Does this work for you guys?